Framework/Geometry/src/MDGeometry/CompositeImplicitFunction.cpp

```diff
@@ -47,10 +49,11 @@ std::string CompositeImplicitFunction::toXMLString() const {
   AutoPtr<Element> parameterListElement = pDoc->createElement("ParameterList");
   functionElement->appendChild(parameterListElement);
 
-  std::string functionXML;
-  for (const auto &Function : m_Functions) {
-    functionXML += Function->toXMLString();
-  }
+  const std::string functionXML =
+      std::accumulate(m_Functions.cbegin(), m_Functions.cend(), std::string(),
```

Comment:
I think `std::accumulate` for string building was on the unclear side of the fence with respect to when STL algorithms should and should not be used.

Reply:
I missed that discussion. I will revert this change. I don't have a strong opinion on this anyway.